### ISSUE TITLE
Rich text processing converts non-breaking spaces into normal spaces

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#4996]
### Issue Summary

When inserting non-breaking spaces into Draftail (with `option + space` on macOS), they are persisted in the database and rendered in the front-end HTML, but subsequently get lost in the DB -> editor JSON conversion.

Initially reported by @philgyford in [Slack #support](https://wagtailcms.slack.com/archives/C81FGJR2S/p1547462877447200).

### Steps to Reproduce

1. Open the editing UI of a page with a rich text field.
2. Type `test test` in the field, with a non-breaking space between the two words
3. Publish the page
4. Look at the front-end HTML of the page – the Chrome DevTools will display the content as `test&nbsp;test`. Inspecting the page shows the character normally, not as an entity.
5. Disable JavaScript in the browser
6. Reload the editing UI of the same page
7. Use the DevTools console to inspect the content of the hidden field. For example, using `JSON.parse(document.querySelector('[name="promo_text"]').value).blocks[0].text.charCodeAt(4)` with the correct `input` selector – this should output 160 for a nbsp, but instead outputs 32 (normal space).

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

Non-breaking spaces should be preserved so they can be used to manage line wrapping.

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes

### Technical details

* Python version: 3.4
* Django version: 2.0
* Wagtail version: 2.5a0
* Browser version: Chrome 71.0.3578.98

---

I suspect this is caused by https://github.com/wagtail/wagtail/blob/057690815b3147be1280834787e275590f71d274/wagtail/admin/rich_text/converters/html_to_contentstate.py#L314-L317

This collapses sequences matching `\s` to ` `.

**Comment by philgyford** [comment]
I wonder if I'm doing something different... if I do steps 1-4, then I _don't_ see a `&nbsp;` in the source of the rendered page. I'm using View Source in Chrome and Safari, not inspecting the element.

This isn't a clean Wagtail install, but I'm on v2.4, Django 2.1.5, Python 3.6.6.

**Comment by philgyford** [comment]
Ah, I took point 4 to mean the opposite of what you meant :)

Short answer: You're right and I've replicated what you described.

But also: It turns out that Safari doesn't show `&nbsp;`s in either View Source _or_ the Elements inspector! This is what lead me to think they were being stripped entirely. If I'd been using Chrome, and seen them in the inspector, I'd have assumed it was all working fine... and then later would have been very confused when they disappeared again!
